contracts/examples/digital-cash/src/digital_cash.rs

```diff
@@ -66,39 +83,77 @@ pub trait DigitalCash {
     ) {
         require!(!self.deposit(&address).is_empty(), "non-existent key");
 
-        let deposit = self.deposit(&address).get();
         let caller_address = self.blockchain().get_caller();
 
-        require!(
-            deposit.expiration_round >= self.blockchain().get_block_round(),
-            "deposit expired"
-        );
-
-        let key = address.as_managed_byte_array();
+        let addr = address.as_managed_byte_array();
         let message = caller_address.as_managed_buffer();
-        require!(
-            self.crypto()
-                .verify_ed25519_legacy_managed::<32>(key, message, &signature),
-            "invalid signature"
-        );
 
-        self.send().direct(
-            &caller_address,
-            &deposit.token_name,
-            deposit.nonce,
-            &deposit.amount,
-        );
-        self.deposit(&address).clear();
+        let mut withdrawed_tokens = ManagedVec::<Self::Api, FundType<Self::Api>>::new();
+        let mut transfer_occured = false;
+        let block_round = self.blockchain().get_block_round();
+        for (key, deposit) in self.deposit(&address).iter() {
+            if deposit.expiration_round >= block_round {
+                require!(
```

Comment:
move this require! out of the FOR function - it is enough to check it only once.